### PR TITLE
add Test for sorting

### PR DESCRIPTION
**Pull request by andmysko**
ADD Cases:
Sort by title from Z to A
sot by orice from lowest to highest
add testIdAttribute
expand the inventory page

### CODE REVIEW

File: src/tests/Sorting.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+test.describe('Sorting Test', () => {
+    test('Sort Item By Title From Z - A', async (
+        /** @type {{ app: import('../pages/Application').Application }} */{ app },
+    ) => {
+        await app.login.navigate();
+        await app.login.performLogin('standard_user', 'secret_sauce');
+        // get default Item list
+        const defaultList = await app.inventory.itemTitle.allTextContents();
+        // sort from A to Z
+        const sortedAscList = defaultList.sort();
```
Comment (EugenePetrik) [inline src/tests/Sorting.spec.js]: We can create a method in page objects which sorts products by specific condition. For example,

```
  /**
   * @param {string} sortBy
   * @param {string[]} productNames
   * @param {number[]} productPrices
   */
  async expectSortedProducts(sortBy, productNames, productPrices) {
    switch (sortBy) {
      case 'Name (A to Z)':
        const sortedNamesAZ = await this.getProductsNames();
        const expectedNamesAZ = productNames.sort();
        expect(sortedNamesAZ).toEqual(expectedNamesAZ);
        break;
      case 'Name (Z to A)':
        const sortedNamesZA = await this.getProductsNames();
        const expectedNamesZA = productNames.sort().reverse();
        expect(sortedNamesZA).toEqual(expectedNamesZA);
        break;
      case 'Price (low to high)':
        const sortedPricesLowHigh = await this.getProductPrices();
        const expectedPricesLowHigh = productPrices.sort((a, b) => a - b);
        expect(sortedPricesLowHigh).toEqual(expectedPricesLowHigh);
        break;
      case 'Price (high to low)':
        const sortedPricesHighLow = await this.getProductPrices();
        const expectedPricesHighLow = productPrices.sort((a, b) => b - a);
        expect(sortedPricesHighLow).toEqual(expectedPricesHighLow);
        break;
      default:
        throw new Error('Sort option is not correct');
    }
  }
```

File: src/tests/Sorting.spec.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        /** @type {{ app: import('../pages/Application').Application }} */{ app },
+    ) => {
+        await app.login.navigate();
+        await app.login.performLogin('standard_user', 'secret_sauce');
+        // get default sorting by Price on UI
+        let defaultListPrice = await app.inventory.itemPrice.allTextContents();
+        // remove $ sign
+        defaultListPrice = defaultListPrice.map((item) => parseInt(item.replace('$', '')));
+        // sort ASC
+        const sortAscPrice = defaultListPrice.toSorted((a, b) => a - b);
```
Comment (EugenePetrik) [inline src/tests/Sorting.spec.js]: Please check the comment above

File: src/tests/Sorting.spec.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        // sort on UI by Price Lo > Hi
+        await app.inventory.sortItemBy('lohi');
+        let SortedPriceAscUi = await app.inventory.itemPrice.allTextContents();
```
Comment (EugenePetrik) [inline src/tests/Sorting.spec.js]: Please follow the naming conventions `let sortedPriceAscUi;`

File: src/tests/Sorting.spec.js (hunk 1 above)
Comment (EugenePetrik) [reply src/tests/Sorting.spec.js]: If you implement this approach, then you can use DDT approach in you tests, for example

```
  const dataSortType = [
    {
      sortBy: 'Name (A to Z)',
    },
    {
      sortBy: 'Name (Z to A)',
    },
    {
      sortBy: 'Price (low to high)',
    },
    {
      sortBy: 'Price (high to low)',
    },
  ];

  dataSortType.forEach(({ sortBy }) => {
    test(`Sort products by ${sortBy}`, async (/** @type {{ app: import('../pages/Application').Application }} */ {
      app,
    }) => {
      const productNames = await app.inventory.getProductsNames();
      const productPrices = await app.inventory.getProductPrices();

      await app.inventory.selectSortOption(sortBy);

      await app.inventory.expectSortedProducts(sortBy, productNames, productPrices);
    });
  });
```

**Review by EugenePetrik — COMMENTED** [review]
Good job! Left a few comments, please check

**Comment by andmysko** [comment]
@EugenePetrik  pls review

**Review by batmanastasiya — APPROVED** [review]

**Pull request merged**